samza-core/src/main/java/org/apache/samza/job/model/JobModelUtil.java

```diff
@@ -19,20 +19,33 @@
 package org.apache.samza.job.model;
 
 import com.google.common.base.Preconditions;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 import org.apache.samza.container.TaskName;
 import org.apache.samza.system.SystemStream;
+import org.apache.samza.SamzaException;
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.serializers.model.SamzaObjectMapper;
 import org.apache.samza.system.SystemStreamPartition;
+import org.codehaus.jackson.map.ObjectMapper;
 
 /**
  * Utility class for the {@link JobModel}
  */
 public class JobModelUtil {
 
+  private static final ObjectMapper MAPPER = SamzaObjectMapper.getObjectMapper();
+
+  private static final int JOB_MODEL_SEGMENT_SIZE_IN_BYTES = 1020 * 1020;
```

Comment:
Zookeeper-Client injects additional metadata on top of the value written to zookeeper node. This allows the space for those additional fields. If we set it to 1024 here, it would still hit the zookeeper-server 1 MB data node limit and fail the job. I was found it during my testing and hence lowered it.

Reply:
It'll be better to add this comment in the code instead of magic numbers.